src/renderer/components/notifications/notifications.store.tsx

```diff
@@ -26,6 +62,29 @@ export class NotificationsStore {
 
   protected autoHideTimers = new Map<NotificationId, number>();
 
+  registerIpcListener(): void {
+    logger.info(`[NOTIFICATION-STORE] start to listen for notifications requests from main`);
+    ipcRenderer.on(NotificationChannelAdd, (event, model: MainNotification) => {
```

Comment:
So either each `iframe` has its own `NotificationStore` (or that used to be the case in 3.X) and we don't want notifications from main to be duplicated or there is only one `NotifcationStore` for all of the `renderer` and this works (without do all the extra work of sending to the other frames). 

Lastly, I don't think that `main` should have to care about `NotificationId` to avoid duplicate rendered notifications.